src/librustc/hir/check_attr.rs

```diff
@@ -118,6 +122,19 @@ impl CheckAttrVisitor<'tcx> {
 
         self.check_repr(item, target);
         self.check_used(item, target);
+
+        // Checks if `#[track_caller]` and `#[naked]` are both used.
+        if let Some(span) = track_caller_span {
+            if item.attrs.iter().any(|attr| attr.check_name(sym::naked)) {
+                struct_span_err!(
+                    self.tcx.sess,
+                    span,
+                    E0901,
+                    "cannot use `#[track_caller]` with `#[naked]`",
+                )
+                .emit();
+            }
+        }
```

Comment:
Can you move this to a new method, e.g. `self.check_track_caller_xor_naked(track_caller_span);`?
Also, this could use https://doc.rust-lang.org/nightly/nightly-rustc/syntax/attr/fn.contains_name.html.

Reply:
I moved the check into `check_track_caller` and I think it makes some sense there, lmk what you think.

Reply:
I like it! :)